linkerd/identity/src/lib.rs

```diff
@@ -318,139 +229,120 @@ impl Crt {
-        f.debug_struct("CrtKey")
-            .field("id", &self.id)
-            .field("expiry", &self.expiry)
-            .finish()
+// === impl LocalId ===
+
+impl From<Name> for LocalId {
+    fn from(n: Name) -> Self {
+        Self(n)
     }
 }
 
-// === impl CertResolver ===
-
-impl rustls::ResolvesClientCert for CertResolver {
-    fn resolve(
-        &self,
-        _acceptable_issuers: &[&[u8]],
-        sigschemes: &[rustls::SignatureScheme],
-    ) -> Option<rustls::sign::CertifiedKey> {
-        // The proxy's server-side doesn't send the list of acceptable issuers so
-        // don't bother looking at `_acceptable_issuers`.
-        self.resolve_(sigschemes)
+impl From<LocalId> for Name {
+    fn from(LocalId(name): LocalId) -> Name {
+        name
     }
+}
 
-    fn has_certs(&self) -> bool {
-        true
+impl AsRef<Name> for LocalId {
+    fn as_ref(&self) -> &Name {
+        &self.0
     }
 }
 
-impl CertResolver {
-    fn resolve_(
-        &self,
-        sigschemes: &[rustls::SignatureScheme],
-    ) -> Option<rustls::sign::CertifiedKey> {
-        if !sigschemes.contains(&SIGNATURE_ALG_RUSTLS_SCHEME) {
-            debug!("signature scheme not supported -> no certificate");
-            return None;
-        }
-        Some(self.0.clone())
+impl fmt::Display for LocalId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
     }
 }
 
-impl rustls::ResolvesServerCert for CertResolver {
-    fn resolve(&self, hello: rustls::ClientHello<'_>) -> Option<rustls::sign::CertifiedKey> {
-        let server_name = if let Some(server_name) = hello.server_name() {
-            server_name
-        } else {
-            debug!("no SNI -> no certificate");
-            return None;
-        };
+// === impl InvalidCrt ===
 
-        // Verify that our certificate is valid for the given SNI name.
-        let c = (&self.0.cert)
-            .first()
-            .map(rustls::Certificate::as_ref)
-            .unwrap_or(&[]); // An empty input will fail to parse.
-        if let Err(err) =
-            webpki::EndEntityCert::from(c).and_then(|c| c.verify_is_valid_for_dns_name(server_name))
-        {
-            debug!(
-                "our certificate is not valid for the SNI name -> no certificate: {:?}",
-                err
-            );
-            return None;
-        }
+// impl fmt::Display for InvalidCrt {
+//     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+//         self.0.fmt(f)
+//     }
+// }
+//
+// impl stdErr::Error for InvalidCrt {
+//     fn source(&self) -> Option<&(dyn stdErr::Error + 'static)> {
+//         self.0.source()
+//     }
+// }
+
+#[derive(Clone)]
+pub struct ClientConfig(pub imp::ClientConfig);
 
-        self.resolve_(hello.sigschemes())
+impl ClientConfig {
+    pub fn empty() -> Self {
+        Self(imp::ClientConfig::empty())
+    }
+    pub fn set_protocols(&mut self, protocols: Vec<Vec<u8>>) {
```

Comment:
nit: prefer whitespace here
```suggestion
    }
    
    pub fn set_protocols(&mut self, protocols: Vec<Vec<u8>>) {
```